File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/EventService.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.management.service;

import io.gravitee.common.data.domain.Page;
import io.gravitee.management.model.EventEntity;
import io.gravitee.management.model.EventQuery;
import io.gravitee.management.model.EventType;
import io.gravitee.management.model.NewEventEntity;

import java.util.Collection;
import java.util.List;
import java.util.Map;
import java.util.function.Function;
import java.util.function.Predicate;

/**
 * @author Titouan COMPIEGNE
 */
public interface EventService {

    EventEntity findById(String id);

    EventEntity create(NewEventEntity event);

    EventEntity create(EventType type, String payload, Map<String, String> properties);
    
    void delete(String eventId);

    Page<EventEntity> search(
            List<EventType> eventTypes, Map<String, Object> properties, long from, long to, int page, int size);

    <T> Page<T> search(List<EventType> eventTypes,
                       Map<String, Object> properties, long from, long to, int page, int size,
                       Function<EventEntity, T> mapper);

    <T> Page<T> search(List<EventType> eventTypes,
                       Map<String, Object> properties, long from, long to, int page, int size,
                       Function<EventEntity, T> mapper, Predicate<T> filter);

    Collection<EventEntity> search(EventQuery query);
}

File: gravitee-apim-console-webui/src/services/event.service.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { IHttpPromise, IHttpService, IPromise } from 'angular';

import { Constants } from '../entities/Constants';
import { Event } from '../entities/event/event';

export class EventService {
  constructor(
    private readonly $http: IHttpService,
    private readonly Constants: Constants,
  ) {}

  public search(type: string, apis: string, from: string, to: string, page: number | string, size: number | string): IHttpPromise<any> {
    return this.$http.get(
      `${this.Constants.env.baseURL}/platform/events?type=${type}&api_ids=${apis}&from=${from}&to=${to}&page=${page}&size=${size}`,
    );
  }

  public findById(apiId: string, eventId: string): IPromise<Event> {
    return this.$http.get<Event>(`${this.Constants.env.baseURL}/apis/${apiId}/events/${eventId}`).then(response => response.data);
  }
}
EventService.$inject = ['$http', 'Constants'];
